docs/madr/decisions/049-policy-on-namespace.md

```diff
@@ -0,0 +1,527 @@
+metadata:
+  namespace: backend-ns
+  labels:
+    kuma.io/origin: zone
+    kuma.io/zone: producer-zone
+spec:
+  targetRef:
+    kind: Mesh
+  to:
+    - targetRef:
+        kind: MeshService
+        labels:
+          kuma.io/display-name: backend
+          k8s.kuma.io/namespace: backend-ns
+---
+# The top-level targetRef references the Mesh, but this is an inbound policy, 
+# and we try to affect requests from all consumers to redis service.
+# This should not be allowed.
+metadata:
+  namespace: backend-ns
+  labels:
+    kuma.io/origin: zone
+    kuma.io/zone: producer-zone
+spec:
+  targetRef:
+    kind: Mesh
+  from:
+    - targetRef:
+        kind: MeshService
+        name: redis
+        namespace: redis-ns
+```
+
+#### Considered options on opting-in(out) to cross-zone policy syncing
+
+* Introduce `spec.export.mode` for all policies, the default is `None`
+* Introduce `spec.export.mode` for all policies, the default is `All`
+* Automatically detect if a policy needs to be exported
+
+##### Decision Outcome
+
+Automatically detect if a policy needs to be exported. 
+Backwards compatibility is tricky, but manageable. 
+We can introduce this behaviour behind the feature flag. 
+This gives users time after the upgrade to make conscious choices on what policies should be exported.
+When the feature flag is switched, Global is going to validate policies against the new enhanced rules.
+
+TODO: how does the Global behave when policy doesn't pass the validation to be synced to other zones?
+
+##### Pros and Cons of introducing `spec.export.mode`, the default is `None`
+
+* Good, because backwards compatible. 
+If the field is not set, the behaviour is similar to what we have in 2.7.
+* Good, because it's the least surprising behaviour. 
+No matter what policy looks like you won't find it affecting pods in other zones. 
+* Bad, because user can forget about it. 
+User assumes MeshHTTPRoute affects all consumers while in reality it affects only consumers in the same zone.
+
+##### Pros and Cons of introducing `spec.export.mode`, the default is `All`
+
+* Good, because coherent with MeshService.
+* Good, because treats all consumers equally by default without favoring the local ones.
+* Bad, because it's not backwards compatible.
+* Bad, because it doesn't add much value.
+It can be easily replaced with the top-level `targetRef{kind:MeshSubset,tags{"kuma.io/zone":"my-zone"}}` when user wants to opt-out.
+
+##### Pros and Cons of automatically detecting if a policy needs to be exported
+
+Having `spec.export.mode` might be excessive taking into account we already have a top-level targetRef.
+If the top-level targetRef specifies a tag `kuma.io/zone` then it's clear whether we need to export the policy or not.
+
+* Good, because we don't introduce new fields, no need for users to learn new concepts.
+* Good, because all consumers are treated equally by default without favoring the local ones.
+* Bad, because it's not backward compatible.
```

Comment:
also bad, because if you want to keep this only to your local zone, you need to specify the name of the zone in the policy